extensions/blocks/eventbrite/eventbrite.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Eventbrite Block.
+ *
+ * @since 8.0.0
+ *
+ * @package Jetpack
+ */
+
+const JETPACK_EVENTBRITE_ID_FROM_URL_REGEX = '(\d+)\/?\s*$';
+const JETPACK_EVENTBRITE_WIDGET_SLUG       = 'eventbrite-widget';
+
+jetpack_register_block(
+	'jetpack/eventbrite',
+	array(
+		'attributes'      => array(
+			'url'      => array(
+				'type' => 'string',
+			),
+			'useModal' => array(
+				'type' => 'boolean',
+			),
+		),
+		'render_callback' => 'jetpack_render_eventbrite_block',
+	)
+);
+
+/**
+ * Eventbrite block registration/dependency delclaration.
+ *
+ * @param array  $attr    Eventbrite block attributes.
+ * @param string $content Rendered embed element (without scripts) from the block editor.
+ *
+ * @return string
+ */
+function jetpack_render_eventbrite_block( $attr, $content ) {
+	if ( empty( $attr['url'] ) ) {
+		return '';
+	}
+
+	$matches = array();
+	preg_match( '/' . JETPACK_EVENTBRITE_ID_FROM_URL_REGEX . '/', $attr['url'], $matches );
+	$event_id = isset( $matches[1] ) && $matches[1] ? $matches[1] : null;
+
+	if ( ! $event_id ) {
+		return '';
+	}
+
+	$widget_id      = JETPACK_EVENTBRITE_WIDGET_SLUG . '-' . $event_id;
+	$no_script_text = __( 'Register on Eventbrite', 'jetpack' );
+
+	wp_enqueue_script( 'eventbrite-widget', 'https://www.eventbrite.com/static/widgets/eb_widgets.js', array(), JETPACK__VERSION, true );
+
+	// Show the embedded version.
+	if ( empty( $attr['useModal'] ) ) {
+		wp_add_inline_script(
+			'eventbrite-widget',
+			"window.EBWidgets.createWidget({
+				widgetType: 'checkout',
+				eventId: ${event_id},
+				iframeContainerId: '${widget_id}',
+			});"
+		);
+
+		return <<<EOT
+${content}
+<noscript>
+	<a href="${attr['url']}" rel="noopener noreferrer" target="_blank">${no_script_text}</a>
+</noscript>
+EOT;
+	}
+
+	// Show the modal version.
+	wp_add_inline_script(
+		'eventbrite-widget',
+		"window.EBWidgets.createWidget({
+			widgetType: 'checkout',
+			eventId: ${event_id},
+			modal: true,
+			modalTriggerElementId: '${widget_id}',
+		});"
+	);
+
+	return <<<EOT
+	<noscript><a href="${attr['url']}" rel="noopener noreferrer" target="_blank"></noscript>
```

Comment:
I don't really understand why we need to break the string down into 2 noscript parts, with one part having only `</a>`. Could you explain things a bit?

Reply:
I've just poked this and I think I've figured it out.

This renders the button style, which basically is a `<button>` that calls some JS on click, to open the event checkout as a modal.

Without JS, the intention is to render the same button, but inside a simple `<a>` linked to the event on Eventbrite.

To avoid printing the button markup twice, once for JS and once for `noscript`, we mix it up a bit:
- `<noscript><a href="$event_url"><noscript>` only renders without JS
- `$block_content` renders always
- `<noscript></a></noscript>` only renders without JS

Reply:
That makes sense, thank you!

> To avoid printing the button markup twice,

I wonder if it would be worth sacrificing that to make the whole thing more readable?

Reply:
The code would be more readable, but the button would be rendered twice when JS is disabled. 😄 

- `$block_content` renders always
- `<noscript><a href="$event_url"> $block_content </a><noscript>` only renders without JS